crates/sui-network/src/state_sync/mod.rs

```diff
@@ -1358,7 +1360,16 @@ where
     let digest = checkpoint.content_digest;
     if let Some(contents) = store
         .get_full_checkpoint_contents_by_sequence_number(*checkpoint.sequence_number())
-        .or_else(|| store.get_full_checkpoint_contents(&digest))
+        .tap_err(|e| debug_fatal!("Failed to get checkpoint contents by sequence number: {e}"))
+        .ok()
+        .flatten()
+        .or_else(|| {
+            store
+                .get_full_checkpoint_contents(&digest)
```

Comment:
From the implementation, it seems we just need to call `get_full_checkpoint_contents(&digest)` once, without calling `get_full_checkpoint_contents_by_sequence_number()`?

Reply:
get_full_checkpoint_contents_by_sequence_number would be loading it from the table directly as an optimization, while get_full_checkpoint_contents I think tries to reconstruct the full contents

Reply:
Oh I think you are right.

Reply:
Ended up removing the get_full_checkpoint_contents_by_sequence_number interface and merged them.